### PR TITLE
Record ONCHAIN orders in contract_observer + a fix or two

### PR DESCRIPTION
**Pull request by JonathonDunford**

### CODE REVIEW

**Comment by JonathonDunford** [comment]
Closes #4

File: app/src/contract_event_recorders.py [hunk 1]
```diff
@@ -182,3 +184,19 @@ def enqueue_order_update_for_transfer(event):
         logger.debug("recorded order cancel signature=%s", signature)
 
     return bool(did_upsert)
+
+# On Order event, record the order, then schedule a job to update potentially affected orders.
```
Comment (freeatnet) [inline app/src/contract_event_recorders.py]: "to update the newly inserted order" would be more accurate

Comment (freeatnet) [inline app/src/contract_event_recorders.py]: "to update the newly inserted order" would be more accurate.

File: app/src/contract_events_backfill.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from ..app import App
 import asyncio
 from ..config import ED_CONTRACT_ADDR, ED_CONTRACT_ABI, HTTP_PROVIDER_URL
-from .contract_event_recorders import record_cancel, record_deposit, process_trade, record_withdraw
+from .contract_event_recorders import record_cancel, record_deposit, record_order, record_trade, record_withdraw
```
Comment (freeatnet) [inline app/src/contract_events_backfill.py]: Should be importing `process_order, process_trade` here.

File: app/src/record_order.py [hunk 3]
```diff
@@ -21,10 +22,18 @@
...
-async def record_order(order):
+async def record_order(order, block_number=0):
     signature = make_order_hash(order)
+    
+    if "r" in order and order["r"] is not None:
+        source = OrderSource.OFFCHAIN
+        date = datetime.utcnow()
+    else:
+        source = OrderSource.ONCHAIN
+        date = datetime.fromtimestamp(block_timestamp(App().web3, block_number), tz=None)
```
Comment (freeatnet) [inline app/src/record_order.py:33]: 👍

File: app/services/contract_observer.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from ..app import App
 import asyncio
-from ..config import ED_CONTRACT_ADDR, ED_CONTRACT_ABI, WS_PROVIDER_URL
-from ..src.contract_event_recorders import record_cancel, record_deposit, process_trade, record_withdraw
+from ..config import ED_CONTRACT_ADDR, ED_CONTRACT_ABI, HTTP_PROVIDER_URL, WS_PROVIDER_URL
+from ..src.contract_event_recorders import record_cancel, record_deposit, record_order, process_trade, record_withdraw
```
Comment (freeatnet) [inline app/services/contract_observer.py]: We should be importing `process_order` here, too.

**Review by freeatnet — COMMENTED** [review]
Some things need attention, but looks on the right track!

**Review by freeatnet — APPROVED** [review]

**Pull request merged**